test/jit/test_isinstance.py

```diff
@@ -0,0 +1,293 @@
+            )
+            assert not torch.jit.isinstance(x, Dict[str, Tuple[int, int, int]])
+            assert not torch.jit.isinstance(x, Tuple[str])
+
+        x = (
+            {"a": ("aa", "aa", "aa"), "b": ("bb", "bb", "bb")},
+            [True, False, True],
+            None,
+        )
+        self.checkScript(tuple_nested, (x,))
+
+    def test_optional_nested(self):
+        def optional_nested(x: Any):
+            assert torch.jit.isinstance(x, Optional[List[str]])
+
+        x = ["a", "b", "c"]
+        self.checkScript(optional_nested, (x,))
+
+    def test_list_tensor_type_true(self):
+        def list_tensor_type_true(x: Any):
+            assert torch.jit.isinstance(x, List[torch.Tensor])
+
+        x = [torch.rand(3, 3), torch.rand(4, 3)]
+        self.checkScript(list_tensor_type_true, (x,))
+
+    def test_tensor_type_false(self):
+        def list_tensor_type_false(x: Any):
+            assert not torch.jit.isinstance(x, List[torch.Tensor])
+
+        x = [1, 2, 3]
+        self.checkScript(list_tensor_type_false, (x,))
+
+    def test_in_if(self):
+        def list_in_if(x: Any):
+            if torch.jit.isinstance(x, List[int]):
+                assert True
+            if torch.jit.isinstance(x, List[str]):
+                assert not True
+
+        x = [1, 2, 3]
+        self.checkScript(list_in_if, (x,))
+
+    def test_if_else(self):
+        def list_in_if_else(x: Any):
+            if torch.jit.isinstance(x, Tuple[str, str, str]):
+                assert True
+            else:
+                assert not True
+
+        x = ("a", "b", "c")
+        self.checkScript(list_in_if_else, (x,))
+
+    def test_in_while_loop(self):
+        def list_in_while_loop(x: Any):
+            count = 0
+            while torch.jit.isinstance(x, List[Dict[str, int]]) and count <= 0:
+                count = count + 1
+            assert count == 1
+
+        x = [{"a": 1, "b": 2}, {"aa": 11, "bb": 22}]
+        self.checkScript(list_in_while_loop, (x,))
+
+    def test_switch_on_type(self):
+        def list_switch_on_type(obj: Any):
+            hit = False
+            if torch.jit.isinstance(obj, List[torch.Tensor]):
+                hit = not hit
+                for el in obj:
+                    # perform some tensor operation
+                    y = el.clamp(0, 0.5)
+            if torch.jit.isinstance(obj, Dict[str, str]):
+                hit = not hit
+                str_cat = ""
+                for val in obj.values():
+                    str_cat = str_cat + val
+                assert "111222" == str_cat
+            assert hit
+
+        x = [torch.rand(3, 3), torch.rand(4, 3)]
+        self.checkScript(list_switch_on_type, (x,))
+        x = {"1": "111", "2": "222"}
+        self.checkScript(list_switch_on_type, (x,))
+
+    def test_list_no_contained_type(self):
+        def list_no_contained_type(x: Any):
+            assert torch.jit.isinstance(x, List)
+
+        x = ["1", "2", "3"]
+
+        try:
```

Comment:
You can use `self.assertRaisesRegex` or `self.assertRaisesRegexWithHighlight` to check for exception instead of writing `try-catch`

Reply:
the self.assertRaisesRegex (from what I can tell) only works with runtime errors generated in the C++ code. Since these errors are generated in Python I wasn't able to get them to work with the self.assertRaisesRegex- do you know if this assumption of mine is wrong?

Reply:
I really are having issues getting self.assertRaisesRegex to work. 

`with self.assertRaisesRegex(
                RuntimeError, 
                "Attempted to use Tuple without a "
                "contained type. Please add a contained type, e.g. "
                "Tuple[int]"
        ):
            torch.jit.script(tuple_no_contained_type`

I'm adding this to the end of the tuple test down below on line 240 and it fails to work. I've tried formatting the error return message string in all the ways I can think of. I also can't find a use of self.assertRaisesRegex which deals with Python-generated errors, the errors they handle in other use cases are all generated from c++ from what I can tell